### PR TITLE
Check for zero people in spatial sweep

### PR DESCRIPTION
**Pull request by njs59**
## Summary

FIx to spatial sweep for if we are trying to infect someone from a group that has no susceptibles left

## Further work

If relevant, add in what needs to be done next on this area (and whether you will do it!)

## Checklist

- [ ] All new functions have docstrings in the correct style
- [ ] I've verified the complete docs build locally without errors
- [ ] I've maintained 100% coverage (please mention any 'no cover' annotations explicitly)
- [ ] I've unit-tested all new methods directly

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] I have read the **CONTRIBUTING** document.
- [ ] My code follows the code style of this project.
- [ ] I have updated the wiki with new parameters/model functionality

## Closing issues

Put `closes #XXXX` in your comment to auto-close the issue that your PR fixes (if such).

### CODE REVIEW

**Review by rccreswell — CHANGES_REQUESTED** [review]
Just need to cover the new lines in testing

File: pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py [hunk 1]
```diff
@@ -150,6 +150,23 @@ def test_find_infectees_fails(self, mock_dist, mock_nan, mock_logger):
         mock_logger.assert_called
         # test logger is called here
 
+    @mock.patch("logging.exception")
+    @mock.patch("numpy.nan_to_num")
+    @mock.patch("pyEpiabm.utility.DistanceFunctions.dist_euclid")
```
Comment (Ellmen) [inline pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py:155]: Do we need these two patches?

**Comment by Ellmen** [comment]
Idea: On line 126, multiply the weights by the population. This makes it more likely that highly populated cells will receive visitors and fixes this issue because a cell with no people will never be selected.

File: pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py [hunk 2]
```diff
@@ -150,6 +150,22 @@ def test_find_infectees_fails(self, mock_dist, mock_nan, mock_logger):
...
+        test_pop = self.pop
+        test_sweep = SpatialSweep()
+        mock_dist.return_value = 2.2
+        test_sweep.bind_population(test_pop)
+
+        self.cell_susc.persons = []
+        test_list = test_sweep.\
+            find_infectees(self.cell_inf, [self.cell_susc], 1)
+        self.assertEqual(test_list, [])
+        mock_logger.assert_called
```
Comment (KCGallagher) [inline pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py:166]: This line is incomplete

**Review by KCGallagher — CHANGES_REQUESTED** [review]
A few small changes. Testing can be tricky - I am happy to talk through some of our tests to explain how mocks and patches etc work, and how we use them

File: pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py (hunk 2 above)
Comment (njs59) [reply pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py]: Removed useless line

File: pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py [hunk 3]
```diff
@@ -150,6 +150,22 @@ def test_find_infectees_fails(self, mock_dist, mock_nan, mock_logger):
         mock_logger.assert_called
         # test logger is called here
 
+    @mock.patch("logging.exception")
+    @mock.patch("pyEpiabm.utility.DistanceFunctions.dist_euclid")
+    def test_find_infectees_fails_2(self, mock_dist, mock_logger):
```
[the remark this answers is not in the document]
Comment (njs59) [reply pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py]: CHanged

File: pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py (hunk 1 above)
Comment (njs59) [reply pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py:150]: All mock patches removed none of them were actually needed

**Review by Ellmen — APPROVED** [review]
This looks good to me. @KCGallagher you can decide if we need to get rid of mocks and/or update the covid-sim changes doc

File: pyEpiabm/pyEpiabm/sweep/spatial_sweep.py [hunk 4]
```diff
@@ -123,7 +123,8 @@ def find_infectees(self, infector_cell: Cell,
         for cell2 in possible_infectee_cells:
             if cell2.id in infector_cell.nearby_cells.keys():
                 distance_weights.append(
-                    1/infector_cell.nearby_cells.get(cell2.id))
+                    len(cell2.persons) /
+                    infector_cell.nearby_cells.get(cell2.id))
```
Comment (KCGallagher) [inline pyEpiabm/pyEpiabm/sweep/spatial_sweep.py:127]: Can you rename `nearby_cells` to `nearby_cell_distances`

**Review by KCGallagher — CHANGES_REQUESTED** [review]
One tiny comment, and could we add a few lines checking the mocks are called in the way we expect (I can do this if needed). Then all good to go

**Comment by KCGallagher** [comment]
The other thing you need to do (and you can merge this first so others can use it if you pinky promise to do it after) is update the "Differences to CovidSim" section of the wiki to include this - we want to describe the issues with how covidsim did it (the old version) and so why we have changed it

**Review by rccreswell — APPROVED** [review]

**Review by KCGallagher — APPROVED** [review]

File: pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py [hunk 5]
```diff
@@ -15,7 +15,7 @@ class TestSpatialSweep(TestMockedLogs):
     def setUp(self):
         # 1st Population
         # 2 cell population
-        # 1st cell: 1 microcell, 100 people, includes the infector
+        # 1st cell: 1 microcell, 1 people, includes the infector
```
Comment (I-Bouros) [inline pyEpiabm/pyEpiabm/tests/test_unit/test_sweep/test_spatial_sweep.py:18]:
```suggestion
        # 1st cell: 1 microcell, 1 person, includes the infector
```

**Review by I-Bouros — CHANGES_REQUESTED** [review]

**Review by I-Bouros — APPROVED** [review]

**Pull request merged**